zio-http/src/main/scala/zio/http/Scheme.scala

```diff
@@ -19,15 +19,14 @@ package zio.http
 import zio.Unsafe
 import zio.stacktracer.TracingImplicits.disableAutoTrace
 
-sealed trait Scheme { self =>
-  def encode: String = self match {
-    case Scheme.HTTP  => "http"
-    case Scheme.HTTPS => "https"
-    case Scheme.WS    => "ws"
-    case Scheme.WSS   => "wss"
-  }
+case class Scheme(value: String) { self =>
```

Comment:
We do never use `case class` as super types in Scala. That leads to unwanted behaviour. Please keep the sealed trait structure and add a new `case class Custom(schema: String)` in the companion object.

Reply:
I will do
Can I ask what type of unwanted behaviours are we talking?

Reply:
this one for example
https://nrinaudo.github.io/scala-best-practices/tricky_behaviours/final_case_classes.html

Reply:
@987Nabil what is we use `final case class` is that ok? I did some more improvements

I experiment have `case class Custom(schema: String)` as an addition to the original code. But after a hour of experimenting I was not liking the result.

Reply:
Please use this structure

```scala
sealed trait Scheme

object Scheme {
  case object HTTP extends Scheme

  case object HTTPS extends Scheme

  case object WS extends Scheme

  case object WSS extends Scheme
  
  case class Custom(scheme: String) extends Scheme
}
```
This is the way we model sum types in Scala. It is the type safe way, that ensures exhaustive pattern match.

If you would model it like you have it right now, you will not see in a pattern match, that the default types are HTTP(S)/WS(S). You will only the a string wrapper.

Reply:
Can you have another look?